scico/denoiser.py

```diff
@@ -25,6 +41,19 @@
     have_bm4d = False
 else:
     have_bm4d = True
+    finfo = np.finfo(np.float32)
+    if hasattr(finfo, "smallest_subnormal"):  # can't test with older numpy versions
+        if finfo.smallest_subnormal == 0.0:
+            warnings.warn(
+                "Importing module bm4d has had an adverse affect on floating point "
+                "accuracy. See the documentation for module scico.denoiser, and "
+                "scico issue #342."
+            )
+            warnings.filterwarnings(
+                action="ignore",
+                message="^The value of the smallest subnormal",
```

Comment:
is this caret character there on purpose?

Reply:
Yes, `message` is a regex.